nix/status-go/default.nix

```diff
@@ -1,28 +1,24 @@
-{ target-os, config, stdenv, callPackage,
-  buildGoPackage, go, fetchFromGitHub, mkFilter, openjdk,
-  androidPkgs, xcodeWrapper }:
+{ config, stdenv, callPackage, mkShell, buildGoPackage, go,
+  fetchFromGitHub, mkFilter, openjdk, androidPkgs, xcodeWrapper }:
 
 let
   inherit (stdenv.lib)
     catAttrs concatStrings fileContents importJSON makeBinPath
-    optional optionalString removePrefix strings attrValues mapAttrs attrByPath
+    optional optionalString strings attrValues mapAttrs attrByPath
     traceValFn;
 
-  platform = callPackage ../platform.nix { inherit target-os; };
-  utils = callPackage ../utils.nix { inherit xcodeWrapper; };
-  gomobile = callPackage ./gomobile { inherit (androidPkgs) platform-tools; inherit target-os xcodeWrapper utils buildGoPackage; };
+  utils = callPackage ./utils.nix { inherit xcodeWrapper; };
+  gomobile = callPackage ./gomobile { inherit (androidPkgs) platform-tools; inherit xcodeWrapper utils buildGoPackage; };
   buildStatusGoDesktopLib = callPackage ./build-desktop-status-go.nix { inherit buildGoPackage go xcodeWrapper utils; };
   buildStatusGoMobileLib = callPackage ./build-mobile-status-go.nix { inherit buildGoPackage go gomobile xcodeWrapper utils; };
   srcData =
     # If config.status_go.src_override is defined, instruct Nix to use that path to build status-go
     if (attrByPath ["status_go" "src_override"] "" config) != "" then rec {
         owner = "status-im";
         repo = "status-go";
-        version = "develop";
```

Comment:
No more `version` for local builds? I'm getting this when using the Go override path:

```
Running: nix-build --pure --fallback --no-out-link --show-trace --attr targets.mobile.android.release --arg config {status_go.src_override="/home/pedro/src/github.com/status-im/status-go";} --arg env {BUILD_ENV="prod";ANDROID_ABI_SPLIT="false";ANDROID_ABI_INCLUDE="armeabi-v7a;arm64-v8a;x86";} --argstr build-type nightly --argstr build-number 9999 --argstr keystore-file /home/pedro/.gradle/status-im.keystore --option extra-sandbox-paths /home/pedro/.gradle/status-im.keystore default.nix --keep TMPDIR,BUILD_ENV,STATUS_GO_SRC_OVERRIDE
error: while evaluating the attribute 'buildPhase' of the derivation 'status-react-build-release-android' at /home/pedro/src/github.com/status-im/status-react/nix/mobile/android/targets/release-android.nix:35:10:
while evaluating the derivation attribute 'name' at /nix/store/id4y2z7gvdb17nhmalqaf4ib11x34nfm-nixpkgs-source/pkgs/stdenv/generic/make-derivation.nix:191:11:
while evaluating the attribute 'version' at /home/pedro/src/github.com/status-im/status-react/nix/status-go/build-status-go.nix:22:5:
attribute 'version' missing, at /home/pedro/src/github.com/status-im/status-react/nix/status-go/default.nix:97:19
```

Reply:
I'm confused as to the purpose of both `version` and `versionName` it's kinda messy. Shouldn't those be combined into one? Or is there any reason not to?

Reply:
`version`: literally what is read from the json file
`sanitizedVersion`: same as `version` but replacing illegal path characters such as `/`
`versionName`: processed version name (`v` prefix removed, changed to `develop` if not a valid version)

Reply:
I've addressed this, please check.